pkg/cloudprovider/aws/ami.go

```diff
@@ -44,17 +45,22 @@ func NewAMIProvider(ssm ssmiface.SSMAPI, clientSet *kubernetes.Clientset) *AMIPr
 	}
 }
 
-func (p *AMIProvider) Get(ctx context.Context, constraints *Constraints) (string, error) {
+func (p *AMIProvider) Get(ctx context.Context, constraints *Constraints, instanceTypes []cloudprovider.InstanceType) (string, error) {
 	version, err := p.kubeServerVersion(ctx)
 	if err != nil {
 		return "", fmt.Errorf("kube server version, %w", err)
 	}
-	var architectureSuffix string
+	var amiNameSuffix string
 	if *constraints.Architecture == v1alpha3.ArchitectureArm64 {
-		architectureSuffix = "-arm64"
+		amiNameSuffix = "-arm64"
 	}
-	// TODO: support for the "amazon-linux-2-gpu" AMI for nvidia use cases
-	name := fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2%s/recommended/image_id", version, architectureSuffix)
+	if NeedsGPUAmi(instanceTypes) {
+		if amiNameSuffix != "" {
+			return "", fmt.Errorf("no amazon-linux-2 ami available for both nvidia gpus and arm64 cpus")
+		}
+		amiNameSuffix = "-gpu"
+	}
+	name := fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2%s/recommended/image_id", version, amiNameSuffix)
```

Comment:
Can you move all of this ami name logic into a separate function?

Reply:
Aye, aye, cap'n

Reply:
Done